pulp_2to3_migration/app/plugin/rpm/repository.py

```diff
@@ -55,24 +55,30 @@ def migrate_to_pulp3(cls, pulp2distributor, repo_version):
             distribution(RpmDistribution): distribution in Pulp 3
             created(bool): True if a distribution has just been created; False if a distribution
                            is an existing one
+
         """
+        pulp2_config = pulp2distributor.pulp2_config
 
         if not repo_version:
             repo_version = pulp2distributor.pulp2_repository.pulp3_repository_version
         publication = repo_version.publication_set.first()
         if not publication:
-            # create publication
-            publish(repo_version.pk)
+            pulp2_checksum_type = pulp2_config.get('checksum_type')
+            checksum_types = None
+            if pulp2_checksum_type:
+                checksum_types = {'metadata': pulp2_checksum_type,
+                                  'package': pulp2_checksum_type}
+            publish(repo_version.pk, checksum_types=checksum_types)
             publication = repo_version.publication_set.first()
 
         # create distribution
-        pulp2_config = pulp2distributor.pulp2_config
         distribution_data = cls.parse_base_config(pulp2distributor, pulp2_config)
 
         # ensure that the base_path does not end with / in Pulp 3, it's often present in Pulp 2.
         base_path = pulp2_config.get('relative_url', pulp2distributor.pulp2_repo_id)
         distribution_data['base_path'] = base_path.rstrip('/')
         distribution_data['publication'] = publication
+        distribution_data['checksum_type'] = pulp2_config.get('checksum_type', '')
```

Comment:
i don't think this is needed, RpmDistribution does nor have a checsum_type field

Reply:
I agree, I believe I removed that. I'll check if it's the recent changes that I pushed :/ thanks